default_warehouse_from_sale_team/models/stock.py

```diff
@@ -27,8 +27,12 @@ def _get_accounting_data_for_valuation(self, move):
         journal_id, acc_src, acc_dest, acc_valuation = super(
             StockQuant, self)._get_accounting_data_for_valuation(move)
 
+        warehouse_valuation = move.warehouse_id.id if move.warehouse_id else \
+            move.picking_type_id.warehouse_id.id if move.picking_type_id else \
+            False
+
         sale_team = self.env['crm.case.section'].search(
-            [('default_warehouse', '=', move.warehouse_id.id)], limit=1)
+            [('default_warehouse', '=', warehouse_valuation)], limit=1)
```

Comment:
you only need  move.warehouse_id remove the .id and is the same no previous validation needed.